Makefile

```diff
@@ -190,7 +190,9 @@ ifeq ("$(TRAVIS_COVERAGE)", "1")
 else
 	@echo "Running in native mode."
 	@export log_level=fatal; export TZ='Asia/Shanghai'; \
-	$(GOTEST) -ldflags '$(TEST_LDFLAGS)' -cover $(PACKAGES) -check.p true -check.timeout 4s || { $(FAILPOINT_DISABLE); exit 1; }
+	set -x
+	$(GOTEST) -timeout 6m -ldflags '$(TEST_LDFLAGS)' -cover $(PACKAGES) -check.p true -check.timeout 4s || { $(FAILPOINT_DISABLE); exit 1; }
```

Comment:
Why we need timeout for `gotest`? For my own laptop, the total test time usually exceeds 6 minutes.

Reply:
The CI environment limits the entire run for 10 minutes and kills the process when it exceeds. Thus it's better to limit the duration in go side so that the running goroutine can be printed for debugging.

Reply:
@glorv Do you have an idea to run the limit in CI enviroment?

Reply:
We can add a custom parameter to add extra parameters to the  test command, like the `$(TEST_LDFLAGS)` do. Buy default it will be empty, and we can add the timeout parameter in ci pipeline

Reply:
@glorv Could you help file a PR?

Reply:
https://github.com/pingcap/tidb/pull/19285 @SunRunAway